File: p008.py
"""Find the greatest product of five consecutive digits in the 1000-digit number.

[p008.txt]

"""
from functools import reduce
from operator import mul
def product(a): return (reduce(mul, a))

digits = [int(digit) for line in open("p008.txt") for digit in line.rstrip()]

ans = max(product(seq5) for seq5 in (digits[i:i+5] for i in range(len(digits) - 4)))

print(ans)

File: p008.scala
/**Find the greatest product of five consecutive digits in the 1000-digit number.

[p008.txt]

*/
val nums = for (
  c <- io.Source.fromFile("p008.txt").mkString
  if c != '\n'
) yield c.asDigit

val products = for {
  i <- 0 to nums.length - 5;
  seq5 = nums.slice(i, i+5)
} yield seq5.product

val ans = products.max

println(ans)
